src/ubx-message.toit

```diff
@@ -532,185 +1577,639 @@ class NavPvt extends Message:
     assert: not payload.is-empty
     return LITTLE-ENDIAN.uint16 payload 90
 
+
 /**
-The UBX-NAV-STATUS message.
+The UBX-NAV-SOL message.
 
-The receiver navigation status.
+Legacy Navigation solution, in ECEF. Included for backwards compatibility.
+  Works on M8 and later however NAV-PVT messages are preferred).
 */
-// https://www.u-blox.com/en/docs/UBX-13003221#%5B%7B%22num%22%3A1204%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C0%2C559.1%2Cnull%5D
-class NavStatus extends Message:
-  /** The UBX-NAV-STATUS message ID. */
-  static ID ::= 0x03
+class NavSol extends Message:
+  static ID ::= 0x06
 
   /** Unknown GNSS fix. */
-  static NO-FIX ::= 0
+  static FIX-TYPE-UNKNOWN ::= 0
   /** Dead reckoning only. */
-  static DEAD-RECKONING-ONLY ::= 1
+  static FIX-TYPE-DEAD ::= 1
   /** 2D fix. */
-  static FIX-2D ::= 2
+  static FIX-TYPE-2D ::= 2
   /** 3D fix. */
-  static FIX-3D ::= 3
-  /** GPS and dead reckoning. */
-  static GPS-DEAD-FIX ::= 4
+  static FIX-TYPE-3D ::= 3
+  /** GNSS and dead reckoning. */
+  static FIX-TYPE-GNSS-DEAD ::= 4
   /** Time only fix. */
-  static TIME-ONLY ::= 5
-  /** Constructs a poll UBX-NAV-STATUS message. */
+  static FIX-TYPE-TIME-ONLY ::= 5
+
+  /** Constructs a poll UBX-NAV-SOL message. */
   constructor.poll:
     super.private_ Message.NAV ID #[]
 
-  constructor.private_ payload:
+  constructor.private_ payload/ByteArray:
     super.private_ Message.NAV ID payload
 
   id-string_ -> string:
-    return "STATUS"
+    return "SOL"
+
+  /** Whether this is a GNSS fix. */
+  is-gnss-fix -> bool:
+    return (flags & 0b00000001) != 0
+
+  /** The time in UTC.
+
+  Time is not included in this legacy Message type.  It can be obtained using
+    other messages however.  Need to think what to do with this one.  Take out
+    for both, or leave in?  Perhaps remove custom properties (made for human
+    consumption) on the message types, and have the driver worry about
+    presentation of message data...
```

Comment:
I would go with that. Keep the UBX messages without too much added functionality.

You could have an additional library next to `ubx-message.toit` (say `util.toit`) that helps with things that come up with ubx messages. If you feel that's useful.